### ISSUE TITLE
Cleanup and optimization

### ISSUE THREAD

**Issue by Eupham** [Eupham/KernelDev#96]
Task1
we've reached a point where things run, mostly clean. Lets fix this minor issue and then I have one more thing for consideration
/usr/local/lib/python3.12/site-packages/torch/optim/lr_scheduler.py:192: UserWarning: Detected call of lr_scheduler.step() before optimizer.step(). In PyTorch 1.1.0 and later, you should call them in the opposite order: optimizer.step() before lr_scheduler.step(). Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
warnings.warn(
This issue is still present despite the prior attempt at fixing it

Task2
Lets do a sanity check, that for the cocktail party task are we using 4 candidates (1 gold, 3 distractors) for each batch member? We're not using any blanks correct?

Task3
I have 2 paths inside of the original_kernel.py triton kernel. I need to ensure both paths are a form of flash attention. Both the teacher forcing path and the cocktail party task.  I don't mean from the model's perspective I mean inside of the kernel.  Don't stop at model.py look inside of original_kernel.py

Task4
I want to keep tokenization on the fly as I'm currently doing but I also want to look for areas we can speed up the overall training without reducing the amount of data we're passing through. I'm also not considering changes to the kernel dimensions/ratios, other than task 3 look elsewhere. So I want to improve the number of iterations per second
